src/Nethermind/Nethermind.Network.Test/PeerManagerTests.cs

```diff
@@ -485,19 +492,67 @@ public void Will_load_static_nodes_and_connect_to_them()
         }
         
         [Test]
-        public void Will_disconnect_on_remove_node()
+        public void Will_disconnect_on_remove_static_node()
         {
             const int nodesCount = 5;
-            var firstSessionDisconnected = false;
+            var disconnections = 0;
             var staticNodes = CreateNodes(nodesCount);
             _staticNodesManager.Nodes.Returns(staticNodes);
             _peerManager.Init();
             _peerManager.Start();
             Thread.Sleep(_travisDelay);
-            _sessions.First().Disconnected += (o, e) => firstSessionDisconnected = true; 
-            _staticNodesManager.NodeRemoved += Raise.EventWith(new RemoveNetworkNodeEventArgs(staticNodes.First()));
+            
+            void DisconnectHandler(object o, DisconnectEventArgs e) => disconnections++;
+            _sessions.ForEach(s => s.Disconnected += DisconnectHandler);
+            
+            _staticNodesManager.NodeRemoved += Raise.EventWith(new NetworkNodeEventArgs(staticNodes.First()));
+            
             _peerManager.ActivePeers.Count(p => p.Node.IsStatic).Should().Be(nodesCount - 1);
-            firstSessionDisconnected.Should().BeTrue();
+            disconnections.Should().Be(1);
+        }
+        
+        [Test]
+        public void Will_connect_and_disconnect_on_peer_management()
+        {
+            var disconnections = 0;
+            _peerManager.Init();
+            _peerManager.Start();
+            var node = new NetworkNode(GenerateEnode());
+            _peerManager.AddPeer(node);
+            Thread.Sleep(_travisDelay);
+
+            _peerManager.ActivePeers.Select(p => p.Node.Id).Should().BeEquivalentTo(new[] {node.NodeId});
+            
+            void DisconnectHandler(object o, DisconnectEventArgs e) => disconnections++;
+            _sessions.ForEach(s => s.Disconnected += DisconnectHandler);
+
+            _peerManager.RemovePeer(node).Should().BeTrue();
+
+            _peerManager.ActivePeers.Should().BeEmpty();
+            disconnections.Should().Be(1);
+        }
+
+        [Test]
+        public void Multiple_addPeer_will_fail_if_peer_already_added()
+        {
+            _peerManager.Init();
+            _peerManager.Start();
+            var node = new NetworkNode(GenerateEnode());
+            _peerManager.AddPeer(node).Should().BeTrue();
+            _peerManager.AddPeer(node).Should().BeFalse();
+            _peerManager.AddPeer(node).Should().BeFalse();
+            Thread.Sleep(_travisDelay);
```

Comment:
I started adding extension methods / loops wherever possible where in a loop I am comparing the results until successful or timeout and then create a one last assertion outside of the loop

it speeds up tests, we can consider it here too

Reply:
I see, I try to use collection asserts where applicable, but this is not needed here